developers/weaviate/starter-guides/custom-vectors.mdx

```diff
@@ -8,187 +8,207 @@ image: og/docs/tutorials.jpg
 import Tabs from '@theme/Tabs';
 import TabItem from '@theme/TabItem';
 import FilteredTextBlock from '@site/src/components/Documentation/FilteredTextBlock';
-import Py4Code from '!!raw-loader!/_includes/code/starter-guides/custom-vectors-py4.py';
-import Py3Code from '!!raw-loader!/_includes/code/starter-guides/custom-vectors-py3.py';
-import Ts3Code from '!!raw-loader!/_includes/code/starter-guides/custom-vectors-ts3.ts';
-import Ts2Code from '!!raw-loader!/_includes/code/starter-guides/custom-vectors-ts2.ts';
+import ByovAllPyCode from '!!raw-loader!/_includes/code/quickstart.byov.all.py';
+import ByovAllPy3Code from '!!raw-loader!/_includes/code/quickstart.byov.all_v3.py';
+import ByovAllTsCode from '!!raw-loader!/_includes/code/quickstart.byov.all.ts';
+import ByovAllTsCodeLegacy from '!!raw-loader!/_includes/code/quickstart.byov.all.legacy.ts';
+import ByovAllShCode from '!!raw-loader!/_includes/code/quickstart.byov.all.sh';
 
 Weaviate is a vector database. [Vector databases](/blog/vector-library-vs-vector-database) store data objects and vectors that represent those objects. When you import a data object, you normally include a vector representation of that object as well. The vector representation is also called an "embedding."
 
-Later, when you work with the stored vectors, Weaviate uses a vectorized version of your query to search the vector space.
+Weaviate has [integrations](/developers/weaviate/model-providers) with [vectorizer](../modules/retriever-vectorizer-modules/index.md) models so you can create vector embeddings at import time. An alternative approach is to "bring your own vectors," and supply the vector embeddings when you upload your data. The embeddings could be custom vectors or pre-existing vectors from one of the popular model providers. This guide discusses importing data that already has vectors.
```

Comment:
Bring the embeddings as an "alternative" approach seems contra to the above re: normal approach.